crates/lib/src/qpu/result_data.rs

```diff
@@ -21,12 +22,25 @@ pub enum ReadoutValues {
     Complex(Vec<Complex64>),
 }
 
+/// A row of data containing the contents of each memory region at the end of a job.
+#[derive(Debug, Clone, EnumAsInner, PartialEq)]
+pub enum MemoryValues {
+    /// Values that correspond to a memory region declared with the BIT or OCTET data type.
+    Binary(Vec<u8>),
+    /// Values that correspond to a memory region declared with the INTEGER data type.
+    Integer(Vec<i64>),
+    /// Values that correspond to a memory region declared with the REAL data type.
+    Real(Vec<f64>),
+}
+
 /// This struct encapsulates data returned from the QPU after executing a job.
 #[allow(clippy::module_name_repetitions)]
 #[derive(Debug, Clone, PartialEq)]
 pub struct QpuResultData {
     pub(crate) mappings: HashMap<String, String>,
     pub(crate) readout_values: HashMap<String, ReadoutValues>,
+    /// The final contents of each memory region, keyed on region name.
+    pub(crate) memory_values: HashMap<String, Option<MemoryValues>>,
```

Comment:
Given that `HashMap::get` already returns an `Option`, I don't follow the need for the value to also be an option. Can't we just omit the key if the value is `None`?

Another note on this down below in the `try_fold`.

Reply:
Yeah, forgot to get ahead of this one with a comment, so thank you for calling it out. I wasn't sure if there was something meaningful about a key appearing with a `None` value versus the key not appearing at all. Since it sounds like there isn't, I'll remove the `Option`.